### PR TITLE
MF-399 - Add open tracing support

### LINKED ISSUE #399: Add message UUID so that we can follow e2e flow
Add UUID to each message that flows though Mainflux so we can log and follow the flow.

For [RewMessage](https://github.com/mainflux/mainflux/blob/master/message.proto#L12) `UUID` field with be UUIDv4. For [Message](https://github.com/mainflux/mainflux/blob/master/message.proto#L21) `UUID` can be `<parent_uuid>.<local_increment>` format - for example `<parent_uuid>.0`,  `<parent_uuid>.1` and so on - because RawMessage is de-aggregated locally in `normalizer`.

### PR DESCRIPTION
**Pull request by anovakovic01**
### What does this do?
Adds base open tracing support to `users` and `things` service.

### Which issue(s) does this PR fix/relate to?
Resolves #399.

### List any changes that modify/break current functionality
There are no such changes.

### Have you included tests for your changes?
Existing tests are updated.

### Did you document any new/modified functionality?
`README.md` doc is updated with new env vars.

### Notes
Open tracing support can be improved by adding tags to the existing spans. Also, open tracing support should be added to bootstrap service. One of the next steps would also be to add tracing of messages that are passing through NATS.

### CODE REVIEW

**Review by drasko — DISMISSED** [review]
LGTM

File: cmd/things/main.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 //
```
Comment (drasko) [inline cmd/things/main.go:1]: I would suggest dropping a year from headers and contracting them to something like this:
```
// Copyright (c) Mainflux
// SPDX-License-Identifier: Apache-2.0
```

File: http/adapter.go [hunk 2]
```diff
@@ -1,5 +1,5 @@
 //
```
Comment (drasko) [inline http/adapter.go:1]: Same here

File: http/api/endpoint.go [hunk 3]
```diff
@@ -1,5 +1,5 @@
 //
```
Comment (drasko) [inline http/api/endpoint.go:1]: And so on...

File: cmd/things/main.go (hunk 1 above)
Comment (anovakovic01) [reply cmd/things/main.go:1]: @drasko can we do this in another PR, this one is already huge?

Comment (drasko) [reply cmd/things/main.go:1]: Yes, sure.

**Comment by mteodor** [comment]
LGTM

**Review by mteodor — APPROVED** [review]

**Review by nmarcetic — APPROVED** [review]

**Pull request merged**